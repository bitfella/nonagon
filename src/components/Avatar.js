import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';

const AvatarWrapper = styled.picture`
  display: block;
  width: 150px;
  height: 150px;
  margin: 0 auto;

  svg {
    width: 100%;
    height: auto;

    polygon {
      stroke: ${props => props.theme.colors.a};
    }
  }
`;

const Avatar = ({ src, alt }) => (
  <AvatarWrapper>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
      aria-label={alt}
    >
      <defs>
        <pattern
          id="img"
          patternUnits="userSpaceOnUse"
          width="100"
          height="100"
        >
          <image href={src} x="6" y="4" width="89" height="89" />
        </pattern>
      </defs>
      <polygon
        fill="url(#img)"
        strokeWidth="2"
        points="50 3.5 79.5 14.5 95.5 42 90 73 66 93.5 34.5 93.5 10 73 5 42 20 15"
      />
    </svg>
  </AvatarWrapper>
);

Avatar.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string
};

export default Avatar;
