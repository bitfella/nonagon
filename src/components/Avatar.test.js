import React from 'react';
import { shallow } from 'enzyme';
import Avatar from './Avatar';

const props = {
  src: 'https://www.google.it'
};

describe('Avatar', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Avatar {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
