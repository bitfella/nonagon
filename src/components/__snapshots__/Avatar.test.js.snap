// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<Avatar__AvatarWrapper>
  <svg
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <pattern
        height="100"
        id="img"
        patternUnits="userSpaceOnUse"
        width="100"
      >
        <image
          height="89"
          href="https://www.google.it"
          width="89"
          x="6"
          y="4"
        />
      </pattern>
    </defs>
    <polygon
      fill="url(#img)"
      points="50 3.5 79.5 14.5 95.5 42 90 73 66 93.5 34.5 93.5 10 73 5 42 20 15"
      strokeWidth="2"
    />
  </svg>
</Avatar__AvatarWrapper>
`;
